File: Organizing Data Java/src/tdtu/edu/OrganizingData/EncapsulateField/Organizing/Student.java
package tdtu.edu.OrganizingData.EncapsulateField.Organizing;

import java.util.Date;

public class Student {
	private String code;
	private String name;
	private Date dateOfBirth;

	public Student(String code, String name, Date dateOfBirth) {
		super();
		this.code = code;
		this.name = name;
		this.dateOfBirth = dateOfBirth;
	}

	public String getCode() {
		return code;
	}

	public void setCode(String code) {
		// check duplicate and valid code 
		this.code = code;
	}

	public String getName() {
		return name;
	}

	public void setName(String name) {
		//check input name is valid
		this.name = name;
	}

	public Date getDateOfBirth() {
		return dateOfBirth;
	}

	public void setDateOfBirth(Date dateOfBirth) {
		// check dateOfBirth
		this.dateOfBirth = dateOfBirth;
	}

}

File: Organizing Data C#/DesignPatternGK/ReplaceRecordwithDataClass/Student.cs
using System;

namespace ReplaceRecordwithDataClass
{
    class Student
    {
        private string _code;
        private string _name;
        private int _age;

        public string Code
        {
            set { _code = value; }
            get { return _code; }
        }

        public string Name
        {
            set { _name = value; }
            get { return _name; }
        }

        public int Age
        {
            set { _age = value; }
            get { return _age; }
        }

        public Student(string code, string name, int age)
        {
            _code = code;
            _name = name;
            _age = age;
        }
    }
}
